presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanCheckerProviderManager.java

```diff
@@ -48,8 +61,38 @@ public void addPlanCheckerProviderFactory(PlanCheckerProviderFactory planChecker
     }
 
     public void loadPlanCheckerProviders()
+            throws IOException
+    {
+        for (File file : listFiles(configDirectory)) {
+            if (file.isFile() && file.getName().endsWith(".properties")) {
+                // unlike function namespaces and connectors, we don't have a concept of catalog
+                // name here (conventionally config file name without the extension)
+                // because plan checkers are never referenced by name.
+                Map<String, String> properties = new HashMap<>(loadProperties(file));
+                checkState(!isNullOrEmpty(properties.get(PLAN_CHECKER_PROVIDER_NAME)),
```

Comment:
```suggestion
                checkArgument(!isNullOrEmpty(properties.get(PLAN_CHECKER_PROVIDER_NAME)),
```

Reply:
i don't think checkArgument is right here because these aren't function arguments

Reply:
We'll often use `checkArgument` in cases where we're just generally evaluating the inputs, and I consider this an input.  I tend to think of `checkState` as being used to identify programming errors where the object is somehow constructed in an incorrect state in an unexpected way.  I looked for examples, but found examples for both, so I'm fine to agree to disagree.

Reply:
makes sense, i guess it is a bit of a grey area. I think I'll leave it as is.